bin/dartdevc.dart

```diff
@@ -57,6 +57,11 @@ Future main(List<String> args) async {
 Future<int> _runCommand(List<String> args,
     {MessageHandler messageHandler}) async {
   try {
+    if (args.isEmpty || args.first != 'compile' && args.first != 'help') {
+      // TODO(jmesserly): we should deprecate the commands. For now they are
+      // still supported for backwards compatibility.
+      args.insert(0, 'compile');
```

Comment:
it may not be obvious from this diff view, but if you expand the file above, you can see that we're guaranteed a fresh mutable list (and there's a comment about it).